erpnext/erpnext_integrations/doctype/leaflink_settings/leaflink_settings.py

```diff
@@ -0,0 +1,182 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+import json
+import pprint
+from frappe.utils import add_days
+import datetime
+import requests
+from frappe.model.document import Document
+import base64
+import hashlib
+import hmac
+
+class LeafLinkSettings(Document):
+	pass
+
+
+@frappe.whitelist()
+def sync_item_with_leaflink(item_details):
+	"""
+	Sync an item in leaflink by hitting the Sync with LeafLink Button.
+	"""
+	pass
+
+
+@frappe.whitelist()
+def sync_customer_with_leaflink(customer_name):
+	"""
+	Sync a customer with leaflink by hitting Sync with Customer Button.
+	"""
+	leaflink_settings = frappe.get_cached_doc("LeafLink Settings")
+	header = {
+		leaflink_settings.get("api_key"): leaflink_settings.get("api_value")
+	}
+
+	#validate if leaflink is enabled
+	if leaflink_settings.get("enable_leaflink") != 1:
+		frappe.throw("Please enable LeafLink in LeafLink Settings.")
+		return
+
+	#check if customer exists on leaflink
+	customers_url = leaflink_settings.get("customers_url")
+	customers_on_leaflink = requests.get(customers_url, headers=header, params={"name": customer_name}).json()
```

Comment:
Lets put this in `try..except` since this request can fail.

Reply:
@rvpasari please make the necessary changes as mentioned by Himanshu.